lnst/Recipes/ENRT/ConfigMixins/EnableIntelSSTTFMixin.py

```diff
@@ -0,0 +1,119 @@
+            return
+
+        for host in self.intel_sst_tf_host_list:
+            if not self._is_sst_supported(host):
+                raise RecipeError(f"Machine {host} doesn't support Intel SST-TF.")
+
+            if not host.run("cpupower idle-set -E").passed:
+                raise RecipeError("Could not enable all idle states.")
+
+            reset_tf_settings = host.run(
+                f"intel-speed-select -c {self._system_cpus(host)} turbo-freq disable –auto"
+            )
+            if not reset_tf_settings.passed:
+                raise RecipeError("Could not reset previous sst-tf settings.")
+
+            set_tf = host.run(
+                f"intel-speed-select -c {self._ssted_cpus()} turbo-freq enable –auto"
+            )
+            if not set_tf.passed:
+                raise RecipeError(f"Could not enable sst-tf for {self._ssted_cpus()}")
+
+    def remove_sub_configuration(self, config):
+        if not self.params.enable_intel_sst_tf:
+            return super().remove_sub_configuration(config)
+
+        logging.warning(
+            "Deconfiguration doesn't work right now. It's a bug in "
+            "intel-speed-select. To deconfigure, reboot the machine..."
+        )
+
+        for host in self.intel_sst_tf_host_list:
+            if not host.run("intel-speed-select turbo-freq disable").passed:
+                logging.warning(
+                    f"Could not remove intel sst-tf configuration on {host}"
+                )
+
+        return super().remove_sub_configuration(config)
+
+    def generate_sub_configuration_description(self, config):
+        description = super().generate_sub_configuration_description(config)
+
+        if self.params.enable_intel_sst_tf:
+            for host in self.intel_sst_tf_host_list:
+                description.append(
+                    f"intel sst-tf enabled on {host} cpus: {self._ssted_cpus()}"
+                )
+        else:
+            description.append("configuration of intel_sst-tf skipped")
+
+        return description
+
+    def _is_sst_supported(self, host):
+        if not is_installed("intel-speed-select"):
+            raise RecipeError("intel-speed-select is not installed.")
+
+        file_check = host.run(f"ls {self.TURBO_PATH}", job_level=ResultLevel.DEBUG)
+        if not file_check.passed:
+            raise RecipeError(
+                "intel-speed-select is not supported." "Not even TurboBoost is"
+            )
+
+        tb_check = host.run(f"cat {self.TURBO_PATH}", job_level=ResultLevel.DEBUG)
+        if not tb_check.passed or tb_check.stdout != "0\n":
+            raise RecipeError("Intel SST-TF requires TurboBoost to be enabled.")
+
+        sst_info = host.run("intel-speed-select --info", job_level=ResultLevel.DEBUG)
+
+        result = sst_info.passed
+        if not re.search(
+            r"Intel\(R\) SST-TF \(feature turbo-freq\) is supported", sst_info.stderr
+        ):
+            result = ResultType.FAIL
```

Comment:
what's the point of this? this is returned from the `is_sst_supported` method but it's only used in a `if` condition in the `apply` method and not added to an actual `Result`... so this could just be a True/False

but maybe you want to do `sst_info.passed = ResultType.FAIL`?